out/notebooks/splunk.ts

```diff
@@ -29,95 +30,157 @@ export function splunkLogin(service) {
         
         service.login(function(err, wasSuccessful) {
             if (err !== null || !wasSuccessful) {
-                reject(err)
+                reject(err);
             } else {
-                resolve(null)
+                resolve(null);
             }
-        })
+        });
 
-    })
+    });
 
 
 }
 
 
 export function createSearchJob(jobs, query, options) {
-
     return new Promise(function(resolve, reject) {
         jobs.create(query, options, function(err, data) {
             if (err !== null) {
                 reject(err);
             } else {
-                resolve(data)
+                resolve(data);
             }
-        })
+        });
+
+    });
+}
 
-    })
+export function dispatchSpl2Module(service: any, spl2Module: string) {
+    // Get first statement assignment '$my_statement = ...' -> 'my_statement' 
+    const statementMatch = spl2Module.match(/\$([a-zA-Z0-9_]+)[\s]*=/m);
+    if (!statementMatch || statementMatch.length < 2) {
+        throw new Error(
+            'No statements found in SPL2. Please assign at least one statement name ' +
+            'using "$". For example: `$my_statement = from _internal`'
+        );
+    }
+    const statementIdentifier = statementMatch[1];
+
+    // The Splunk SDK for Javascript doesn't currently support the spl2-module-dispatch endpoint
+    // nor does it support sending requests in JSON format (only receiving responses), so
+    // for now use the underlying needle library that the SDK uses for requests/responses
+    return new Promise(function(resolve, reject) {
+        needle.request(
```

Comment:
What version of needle are we using? Need 2.0 support then chaining. https://github.com/tomas/needle#readme

Reply:
Oh good call! Much cleaner